### PR TITLE
Ensure single trace pipeline

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

- Implement the lock mechanism that ensures that only one trace pipeline can be active (described in https://github.com/kyma-project/community/pull/748)
- Move the reason strings from the API pkg to the respective controllers, similar to what Kubernetes does

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/15836

### CODE REVIEW

File: components/telemetry-operator/controller/tracepipeline/controller.go [hunk 1]
```diff
@@ -169,3 +177,37 @@ func (r *Reconciler) doReconcile(ctx context.Context, pipeline *telemetryv1alpha
...
+	if err := r.Get(ctx, lockName, &lock); err != nil {
+		if apierrors.IsNotFound(err) {
+			return r.createLock(ctx, lockName, pipeline)
+		}
+		return fmt.Errorf("failed to get lock: %v", err)
+	}
+
+	for _, ref := range lock.GetOwnerReferences() {
+		if ref.Name == pipeline.Name && ref.UID == pipeline.UID {
+			// I am the owner
```
Comment (lindnerby) [inline components/telemetry-operator/controller/tracepipeline/controller.go:193]: Can be removed

**Review by lindnerby — CHANGES_REQUESTED** [review]

File: components/telemetry-operator/controller/tracepipeline/controller.go (hunk 1 above)
Comment (skhalash) [reply components/telemetry-operator/controller/tracepipeline/controller.go]: True

**Review by lindnerby — APPROVED** [review]

**Review by lindnerby — APPROVED** [review]

**Pull request merged**